packages/ariakit/src/combobox/__examples__/combobox-typing-only-popover/index.tsx

```diff
@@ -0,0 +1,47 @@
+import {
+  Combobox,
+  ComboboxItem,
+  ComboboxPopover,
+  useComboboxState,
+} from "ariakit/combobox";
+import "./style.css";
+
+export default function Example() {
+  const combobox = useComboboxState({
+    gutter: 8,
+    sameWidth: true,
+  });
+
+  return (
+    <div>
+      <label className="label">
+        Your favorite fruit
+        <Combobox
+          state={combobox}
+          placeholder="e.g., Apple"
+          className="combobox"
+          showOnMouseDown={false}
+          showOnKeyDown={false}
```

Comment:
Instead of using those props, you can just do this, which results in better behavior:

```suggestion
  const combobox = useComboboxState({
    gutter: 8,
    sameWidth: true,
  });
  
  // It's safe to derive state here as long as we're not overwriting
  // combobox.setOpen
  if (combobox.open && combobox.value.length < 1) {
    combobox.setOpen(false);
  }

  return (
    <div>
      <label className="label">
        Your favorite fruit
        <Combobox
          state={combobox}
          placeholder="e.g., Apple"
          className="combobox"
```

Reply:
Of course, if you intend to create a "typing-only" combobox, your solution is more accurate. My suggestion only applies to "open popover only after one character typed".

Reply:
I'm new to the repo, and I'm not aware of the internals, so, calling setOpen does not count as a new render there?

Reply:
That's not an internal thing. That's [the functional version of `getDerivedStateFromProps`](https://reactjs.org/docs/hooks-faq.html#how-do-i-implement-getderivedstatefromprops). That will render twice, but one of the renders will be discarded by React. Discarded renders don't advance to the commit phase and don't run effects, they are usually cheap.